src/libAtomVM/atom_table.c

```diff
@@ -445,26 +445,68 @@ long atom_table_ensure_atom(struct AtomTable *table, AtomString string, enum Ato
     return new_index;
 }
 
-int atom_table_ensure_atoms(
-    struct AtomTable *table, const void *atoms, int count, int *translate_table)
+static inline int read_encoded_len(const uint8_t **len_bytes)
 {
+    uint8_t byte0 = (*len_bytes)[0];
+
+    if ((byte0 & 0x8) == 0) {
+        (*len_bytes)++;
+        return byte0 >> 4;
+
+    } else if ((byte0 & 0x10) == 0) {
+        uint8_t byte1 = (*len_bytes)[1];
+        (*len_bytes) += 2;
+        return ((byte0 >> 5) << 8) | byte1;
+
+    } else {
+        return -1;
+    }
+}
+
+int atom_table_ensure_atoms(struct AtomTable *table, const void *atoms, int count,
+    int *translate_table, enum EnsureAtomsOpt opt)
+{
+    bool is_long_format = (opt & EnsureLongEncoding) != 0;
+
     SMP_WRLOCK(table);
 
     int new_atoms_count = 0;
 
     const uint8_t *current_atom = atoms;
 
     for (int i = 0; i < count; i++) {
-        struct HNode *node = get_node(table, current_atom);
+        struct HNode *node;
+        if (is_long_format) {
+            int atom_len = read_encoded_len(&current_atom);
+            if (atom_len < 0) {
+                fprintf(stderr, "Found invalid atom len.");
+                AVM_ABORT();
+            } else if (atom_len > 255) {
+                fprintf(stderr,
+                    "Unsupported %i bytes atom:\n"
+                    "AtomVM unlike OTP >= 28 doesn't support atoms containing up to 255"
+                    "codepoints, but it is limited to 255 bytes.\n"
+                    "If you see this unlikely error please let the developers know.\n",
```

Comment:
Help me rephrase this, I can write it in a better way.

Reply:
Maybe:
"Unsupported AtomVM length %i bytes.\n"
"Unlike OTP >= 28, AtomVM supports a maximum of 255 codepoints.\n"
"If you are seeing this error please open an issue on GitHub:\n"
"https://github.com/atomvm/AtomVM/issues\n"